internal/controller/cluster_controller.go

```diff
@@ -1561,3 +1709,96 @@ func (r *ClusterReconciler) markPVCReadyForCompletedJobs(
 
 	return nil
 }
+
+// checkForEquilibriumState detects when the cluster is stuck in an equilibrium state
+// where jobs are running but making no progress, potentially due to missing PVCs or other issues
+func (r *ClusterReconciler) checkForEquilibriumState(
+	ctx context.Context,
+	cluster *apiv1.Cluster,
+	resources *managedResources,
+) error {
+	contextLogger := log.FromContext(ctx)
+
+	// Check if we have long-running jobs that might be stuck
+	equilibriumTimeout := 15 * time.Minute // Consider equilibrium after 15 minutes
+
+	for _, job := range resources.jobs.Items {
+		// Skip completed or failed jobs
+		if utils.IsJobComplete(job) || utils.IsJobFailed(job) {
+			continue
+		}
+
+		// Check if job has been running for too long without progress
+		if job.CreationTimestamp.Add(equilibriumTimeout).Before(time.Now()) {
+			// Check if job has no active pods (stuck in pending)
+			if job.Status.Active == 0 && job.Status.Succeeded == 0 && job.Status.Failed == 0 {
```

Comment:
If the job is failing to schedule for a long time, I'd be more inclined to call it stuck and consider it a failure than if it gets scheduled.